packages/pass-style/src/symbol.js

```diff
@@ -1,117 +1,70 @@
-  if (name === undefined) {
-    return wellKnownSymbolNames.get(sym);
-  }
-  if (name.startsWith('@@')) {
-    return `@@${name}`;
-  }
-  return name;
-};
+export const nameForPassableSymbol = sym =>
+  isPassableSymbol(sym) ? sym.description : undefined;
 harden(nameForPassableSymbol);
 
-const AtAtPrefixPattern = /^@@(.*)$/;
-harden(AtAtPrefixPattern);
+const symbolAsyncIteratorDescription = Symbol.asyncIterator.description;
+
+const specialCaseAsyncIteratorSymbol =
+  getEnvironmentOption('PASS_STYLE_LEGACY_ASYNC_ITERATOR_SYMBOL', 'disabled', [
+    'enabled',
+  ]) === 'enabled';
 
 /**
- * If `name` is a string that could have been produced by
- * `nameForPassableSymbol`, return the symbol argument it was produced to
- * represent.
+ * Return a *fresh* unregistered not-well-known symbol whose `description`
+ * is `name`.
  *
- *    If `name` does not begin with `"@@"`, then just the corresponding
- *      registered symbol, `Symbol.for(name)`.
- *    If `name` is `"@@"` followed by a well known symbol's property name on
- *      `Symbol` such `"@@iterator", return that well known symbol such as
- *      `Symbol.iterator`
- *    If `name` begins with `"@@@@"` it encodes the registered symbol whose
- *      name begins with `"@@"` instead.
- *    Otherwise, if name begins with `"@@"` it may encode a registered symbol
- *      from a future version of JavaScript, but it is not one we can decode
- *      yet, so throw.
+ * Alternatively, to soften the transition from legacy,
+ * if environment variable `PASS_STYLE_LEGACY_ASYNC_ITERATOR_SYMBOL`
+ * is `'enabled'` (not the default) and the name is either `'@@asyncIterator'`
+ * or the `description` string of `Symbol.asyncIterator`, then return
+ * `Symbol.asyncIterator`.
+ *
+ * TODO Once we're on the other side of this transition, we will eventually
+ * stop supporting this config switch.
  *
  * @param {string} name
- * @returns {symbol=}
+ * @returns {symbol}
  */
-export const passableSymbolForName = name => {
-  if (typeof name !== 'string') {
-    return undefined;
-  }
-  const match = AtAtPrefixPattern.exec(name);
-  if (match) {
-    const suffix = match[1];
-    if (suffix.startsWith('@@')) {
-      return Symbol.for(suffix);
-    } else {
-      const sym = Symbol[suffix];
-      if (typeof sym === 'symbol') {
-        return sym;
-      }
-      Fail`Reserved for well known symbol ${q(suffix)}: ${q(name)}`;
-    }
-  }
-  return Symbol.for(name);
-};
+export const passableSymbolForName = name =>
+  specialCaseAsyncIteratorSymbol &&
+  ['@@asyncIterator', symbolAsyncIteratorDescription].includes(name)
+    ? Symbol.asyncIterator
+    : Symbol(name);
 harden(passableSymbolForName);
+
+/**
+ * An adapter to help transition to flip which symbols are passable.
+ * See `ses-utils.js` in `@agoric/internal` at
+ * https://github.com/Agoric/agoric-sdk/pull/11338
+ *
+ * @param {string} name
+ * @returns {symbol}
+ */
+export const unpassableSymbolForName = name => Symbol(name);
```

Comment:
name is confusing

Reply:
Suggestions?

Reply:
Huh. Now I'm confused. Hold on. More soon.

Reply:
The right hand side should have been `Symbol.for(name)`. Fixed.